index.js

```diff
@@ -486,4 +483,31 @@ FeatureService.prototype._catchErrors = function (task, e, url, cb) {
   }.bind(this), task.retry * 1000)
 }
 
+/* Forms a standardized error object for reporting back up the chain
+ * @param {object} error - http request error
+ * @param {string} message - human readable message about ther
+ * @param {object} json - json returned from server request
+ * @returns {object} - error object containing diagnostic information
+ */
+FeatureService.prototype._buildError = function (url, message, error) {
+  if (!error) error = {}
+  error.message = message
+  error.request = url
+  error.response = error.message
+  error.type = error.type || 'Unknown failure'
+  error.code = error.code || 500
+
+  return error
+}
```

Comment:
To follow up, we really do need this additional error information. Otherwise requesting clients won't know what actually went wrong.

In Open Data, our use case is alerting the owners of the servers about which requests failed.